src/init.rs

```diff
@@ -11,29 +12,46 @@ rest of the script, and you have to spam semicolons everywhere. By using
 source and process substitutions, we make it possible to comment and debug
 the init scripts. */
 
+fn get_starship_path() -> io::Result<String> {
+    let current_exe = env::current_exe()?
+        .to_str()
+        .ok_or_else(|| io::Error::new(io::ErrorKind::Other, "can't convert to str"))?
+        .to_string();
+    Ok(current_exe)
+}
+
 /* This prints the setup stub, the short piece of code which sets up the main
 init code. The stub produces the main init script, then evaluates it with
 `source` and process substitution */
-pub fn init_stub(shell_name: &str) {
+pub fn init_stub(shell_name: &str) -> io::Result<()> {
```

Comment:
A little confused by this change. Is it just stylistic, or does it enable something in the code that we didn't have before?

Reply:
(and same question for `init_main`)

Reply:
Normally I try to avoid using `unwrap()` an `expect()` and I prefer to use `Result` everywhere. But I could use `expect()` on `path_to_starship()`. I guess it shouldn't fail.

Reply:
Oh, I see, this is for error handling outside the function.

Keep it then. We need to start removing places where things panic, and moving to `Result` and `Option` is a good step in that direction. I was just wondering if there was something else I was missing.

Reply:
Gotcha.